Database/Entities/Creature.cs

```diff
@@ -1,24 +1,82 @@
 ﻿namespace Database.Entities;
 
+/// <summary>
+/// Represents an in-game creature and it's data.
+/// Creatures belong to a specific faction and are categorized by tiers, with Tier 1 being the weakest 
+/// and Tier 7 being the strongest. <para/>
+/// 
+/// The <c>Creature</c> class provides properties to store and update the unique identifier, faction id,
+/// tier, base cost, upgrade cost, growth and image data for it's base, upgraded and alternate forms. <para/>
+/// 
+/// The binary image data stored in <c>ImageBytesBase</c>, <c>ImageBytesUpg</c> and <c>ImageBytesUpgAlt</c> 
+/// can be converted to a visual representation using the <c>System.Drawing.Image</c> class. <br/>
+/// Note that <c>System.Drawing</c> might require additional dependencies on some platforms. <para/>
+/// 
+/// <b>Additionally</b>, it includes a <c>Faction</c> entity to which it belongs to.
+/// </summary>
 public partial class Creature
 {
+    /// <summary>
+    /// Gets or sets the unique identifier.
+    /// This is a 32-bit signed integer.
+    /// </summary>
     public int ID { get; set; }
 
+    /// <summary>
+    /// Gets or sets the identifier of the faction the creature belongs to.
+    /// This is a 32-bit signed integer.
+    /// </summary>
     public int FactionID { get; set; }
 
+    /// <summary>
+    /// Gets or sets the raw binary image data for the creature's base form. 
+    /// <i>For example, a Peasant.</i><para/>
+    /// This can be used to reconstruct the image using libraries such as <c>System.Drawing</c>.
+    /// </summary>
     public byte[] ImageBytesBase { get; set; }
 
+    /// <summary>
+    /// Gets or sets the raw binary image data for the creature's first upgraded form. 
+    /// <i>For example, a Peasant (base form) upgrades to a Militiaman.</i> <para/>
+    /// This can be used to reconstruct the image using libraries such as <c>System.Drawing</c>.
+    /// </summary>
     public byte[] ImageBytesUpg { get; set; }
 
+    /// <summary>
+    /// Gets or sets the raw binary image data for the creature's first alternate upgraded form. <para/>
+    /// Alternate upgrades are introduced in the <b>Tribes of the East</b> expansion. 
+    /// <i>For example, a Peasant (base form) upgrades to a Brute.</i> <para/>
+    /// This can be used to reconstruct the image using libraries such as <c>System.Drawing</c>.
+    /// </summary>
```

Comment:
Apart from the previous comments, line 47 introduces information completely irrelevant to the entire project.
In addition your previous summary (line 39 specifically) alluded to there being, at least, a "second" upgraded form but it is nowhere to be found. Similarly, line 46 alludes to there being, at least, a "second" alternate upgraded form.

Better alternative would be for the summary of `ImageBytesUpg` to be something akin to

> Byte representation of the image of the creature's base upgrade.

While this summary to read something like:

> Byte representation of the image of the creature's alternative upgrade.

"Upgraded form" could be used but in practice (at least from what I've seen and experienced) people refer to the upgraded forms simply as "upgrades".